src/diff/index.js

```diff
@@ -433,7 +433,9 @@ function diffElementNodes(
 				// despite the attribute not being present. When the attribute
 				// is missing the progress bar is treated as indeterminate.
 				// To fix that we'll always update it when it is 0 for progress elements
-				(i !== dom.value || (nodeType === 'progress' && !i))
+				(i !== oldProps.value ||
```

Comment:
This fixes a bug in IE11 where we never apply the value prop for `<option value="X">X<option>`. In that browser, `option.value` is `X` as inferred from the option's text content, however inferred option values can't be selected via `select.value`, so we need to assign to it manually.

This also fixes a potential issue where the text content of an `<option>` is initially some value, but then later changes - if we skip assigning `option.value`, the value-to-text binding remains live instead of being correctly locked to the `value` prop from VDOM.